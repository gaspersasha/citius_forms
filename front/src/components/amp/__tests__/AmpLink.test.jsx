import React from 'react';
import { AmpLink } from '..';

const defaultProps = {
  isAmp: false,
  link: 'test',
  children: 'test',
};

describe('<AmpLink />', () => {
  let wrapper;

  const getWrapper = (props) =>
    shallow(<AmpLink {...defaultProps} {...props} />);

  it('renders without errors', () => {
    wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it('handles amp case', () => {
    wrapper = getWrapper({ isAmp: true });

    expect(wrapper.html()).toEqual(
      '<a on="tap:AMP.navigateTo(url=&#x27;test&#x27;)">test</a>'
    );
  });
});
