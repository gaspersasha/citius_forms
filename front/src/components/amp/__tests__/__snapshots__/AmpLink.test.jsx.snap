// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmpLink /> renders without errors 1`] = `
<Fragment>
  <a
    href="test"
  >
    test
  </a>
</Fragment>
`;
